x-pack/test/oidc_api_integration/apis/security/oidc_implicit_flow.ts

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import { JSDOM } from 'jsdom';
+import request, { Cookie } from 'request';
+import { createTokens, getStateAndNonce } from '../../fixtures/oidc_tools';
+import { KibanaFunctionalTestDefaultProviders } from '../../../types/providers';
+
+// eslint-disable-next-line import/no-default-export
+export default function({ getService }: KibanaFunctionalTestDefaultProviders) {
+  const supertest = getService('supertestWithoutAuth');
+
+  describe('OpenID Connect Implicit Flow authentication', () => {
+    describe('finishing handshake', () => {
+      let stateAndNonce: ReturnType<typeof getStateAndNonce>;
+      let handshakeCookie: Cookie;
+
+      beforeEach(async () => {
+        const handshakeResponse = await supertest
+          .get('/abc/xyz/handshake?one=two three')
+          .expect(302);
+
+        handshakeCookie = request.cookie(handshakeResponse.headers['set-cookie'][0])!;
+        stateAndNonce = getStateAndNonce(handshakeResponse.headers.location);
+      });
+
+      it('should return an HTML page that will parse URL fragment', async () => {
```

Comment:
__note:__ aaaand here is the result of my anti-desire to write functional/browser tests for this piece of functionality :see_no_evil: Please, let me know if you don't like this and rather have functional/browser test.